### PR TITLE
Add spectrum module

### PR DESCRIPTION
**Pull request by alexmaragko**
- Spectrum module and tests added. 
- Mock fitted.py module added.
- Example galaxy spectrum for fitting usage added in the resources directory.

### CODE REVIEW

File: amespahdbpythonsuite/spectrum.py [hunk 1]
```diff
@@ -1,14 +1,147 @@
...
     def __init__(self, d=None, **keywords):
         Transitions.__init__(self, d, **keywords)
-        return None
+
+    def set(self, d=None, **keywords):
+        """
+        Calls class: :class:`amespahdbpythonsuite.transitions.Transitions.set to parse keywords.
+
+        """
+        Transitions.set(self, d, **keywords)
```
Comment (PAHdb) [inline amespahdbpythonsuite/spectrum.py]: This line probably doesn't need to be here when line 36 is moved out of the nested if-statements.

File: amespahdbpythonsuite/spectrum.py [hunk 2]
```diff
@@ -1,14 +1,147 @@
...
+        if d:
+            if d.get('type', '') == self.__class__.__name__:
+                if not keywords.get('grid'):
+                    self.grid = d['grid']
+                if not keywords.get('profile'):
+                    self.profile = d['profile']
+                if not keywords.get('fwhm'):
+                    self.fwhm = d['fwhm']
+
+                Transitions.set(self, d, **keywords)
```
Comment (PAHdb) [inline amespahdbpythonsuite/spectrum.py]: Remove line 26 and indent this line out of the nested if-statements.

File: amespahdbpythonsuite/spectrum.py [hunk 3]
```diff
@@ -1,14 +1,147 @@
...
+        """
+        d = Transitions.get(self)
+        d['type'] = self.__class__.__name__
+        d['grid'] = self.grid
+        d['profile'] = self.profile
+        d['fwhm'] = self.fwhm
+
+        return d
+
+    def subdatabase(self, uids):
```
Comment (PAHdb) [inline amespahdbpythonsuite/spectrum.py]: This method is not needed.

File: amespahdbpythonsuite/spectrum.py [hunk 4]
```diff
@@ -1,14 +1,147 @@
...
+                data[uid] = np.array(intensities)
+                weights[uid] = s
+
+        # Call Fitted Class to plot the fitted spectrum and get the fit breakdown.
+        from amespahdbpythonsuite.fitted import Fitted
+
+        return Fitted(type=self.type,
+                      version=self.version,
+                      data=data,
+                      pahdb=self.subdatabase(uids),
```
Comment (PAHdb) [inline amespahdbpythonsuite/spectrum.py]: Passing self.pahdb is sufficient here--its done by reference, so this wouldn't take up any additional memory.

File: amespahdbpythonsuite/tests/test_spectrum.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+@pytest.fixture(scope="module")
+def pahdb_theoretical():
+    xml = 'resources/pahdb-theoretical_cutdown.xml'
+    pahdb = AmesPAHdb(filename=resource_filename('amespahdbpythonsuite', xml),
+                      check=False, cache=False)
+    return pahdb
+
+
+class TestSpecies():
```
Comment (PAHdb) [inline amespahdbpythonsuite/tests/test_spectrum.py]: Class name should probably be TestSpectrum?

File: amespahdbpythonsuite/tests/test_spectrum.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        # convolve the transitions into a spectrum.
+        spectrum = transitions.convolve(grid=waven, fwhm=15.0, gaussian=True, multiprocessing=False)
+        # fit the spectrum.
+        fit = spectrum.fit(flux, sigma)
+        # Assert results.
+        assert spectrum.uids == uids
+        assert fit.uids == [73, 2054, 223]
+        np.testing.assert_array_almost_equal(fit.grid, np.array(waven))
+        # Check plotting function.
+        monkeypatch.setattr(plt, 'show', lambda: None)
```
Comment (PAHdb) [inline amespahdbpythonsuite/tests/test_spectrum.py:59]: Didn't know about monkeypatch-pretty cool!

**Review by PAHdb — CHANGES_REQUESTED** [review]

**Comment by alexmaragko** [comment]
Updated to address the PR comments.

**Review by PAHdb — APPROVED** [review]

**Pull request merged**